be/src/storage/lake/lake_primary_index.cpp

```diff
@@ -168,4 +175,60 @@ Status LakePrimaryIndex::_do_lake_load(TabletManager* tablet_mgr, const TabletMe
     return Status::OK();
 }
 
+Status LakePrimaryIndex::apply_opcompaction(const TabletMetadata& metadata,
```

Comment:
if (!_enable_persistent_index) {
  // do nothing
} else if (metadata.persistent_index_type() == PersistentIndexTypePB::LOCAL) {
  // do nothing
}  else if (metadata.persistent_index_type() == PersistentIndexTypePB::CLOUD_NATIVE) {
   if (lake_persistent_index == nullptr) {
       return error.
   }
   return dynamic_cast<LakePersistentIndex*>(_persistent_index.get())->apply_opcompaction(op_compaction);
} else {
   return Status::InternalError("Persistent index is invalid.");
}